challenges/no-repeats.js

```diff
@@ -0,0 +1,31 @@
+function permAlone(str) {
```

Comment:
whew this method is a mess!  I look forward to some further PRs breaking down into modular functions.